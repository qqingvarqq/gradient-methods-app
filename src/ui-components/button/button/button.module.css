.root {
  background-color: var(--accent-color);
  color: var(--secondary-font-color);
  width: fit-content;
  font-size: 24px;
  height: 48px;
  border-radius: 24px;
  padding: 0px 16px 0px 16px;
  box-sizing: border-box;
}
.secondary {
  background-color: var(--primary-color);
  color: var(--primary-font-color);
  transition: box-shadow 0.3s ease;
}
.secondary:hover {
  box-shadow: 0px 0px 0px 1px var(--accent-color);
}
