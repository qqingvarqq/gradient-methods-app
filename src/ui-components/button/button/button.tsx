import * as React from 'react';
import styles from './button.module.css';
import {BaseButton, BaseButtonProps} from '../base-button/base-button';

export const Button: React.FC<
  BaseButtonProps & {
    secondary?: boolean;
  }
> = (props) => {
  const {className = '', secondary, ...rest} = props;

  return (
    <BaseButton
      className={`${styles.root} ${secondary && styles.secondary} ${className}`}
      {...rest}
    />
  );
};
