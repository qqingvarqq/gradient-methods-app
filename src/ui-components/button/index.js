export {BaseButton} from './base-button';
export {IconButton} from './icon-button';
export {SubmitButton} from './submit-button';
