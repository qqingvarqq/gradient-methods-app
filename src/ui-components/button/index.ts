export * from './base-button/base-button';
export * from './icon-button/icon-button';
export * from './button/button';
