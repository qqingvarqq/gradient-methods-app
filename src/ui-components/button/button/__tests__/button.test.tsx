import * as React from 'react';
import {render, fireEvent} from '@testing-library/react';
import {Button} from '../button';

describe('Button', () => {
  it('should render without crashing', () => {
    const {asFragment} = render(<Button>btn</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render secondary button', () => {
    const {asFragment} = render(<Button secondary> btn</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should be clickable', () => {
    const onClick = jest.fn();
    const {container} = render(<Button onClick={onClick} />);
    // @ts-ignore
    fireEvent.click(container.firstChild);
    expect(onClick).toBeCalled();
  });
});
