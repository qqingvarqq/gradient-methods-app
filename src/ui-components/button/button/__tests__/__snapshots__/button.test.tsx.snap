// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render secondary button 1`] = `
<DocumentFragment>
  <button
    class="root root secondary "
  >
     btn
  </button>
</DocumentFragment>
`;

exports[`Button should render without crashing 1`] = `
<DocumentFragment>
  <button
    class="root root undefined "
  >
    btn
  </button>
</DocumentFragment>
`;
